### PR TITLE
[DLS-7863] change list returns date to end of tax year

### PR DESCRIPTION
**Pull request by nubz**

### CODE REVIEW

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/services/returns/ReturnsService.scala [hunk 1]
```diff
@@ -611,7 +611,7 @@ class ReturnsServiceImpl @Inject() (
   ): EitherT[Future, Error, List[ReturnSummary]] = {
     val today    = LocalDate.now()
     val fromDate = today.minusYears(viewConfig.numberOfTaxYearsForReturns).plusDays(1L)
-    val toDate   = today
+    val toDate   = TimeUtils.getTaxYearEndDateInclusive(today)
```
Comment (LAJW) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/services/returns/ReturnsService.scala:614]: I'd put a comment here (or in the connector) explaining why we're supplying toDate as the end of the tax year (and that it wouldn't work with LocalDate.now()).

File: test/uk/gov/hmrc/cgtpropertydisposalsfrontend/services/returns/ReturnsServiceImplSpec.scala [hunk 2]
```diff
@@ -1875,8 +1875,9 @@ class ReturnsServiceImplSpec extends AnyWordSpec with Matchers with MockFactory
     "handling requests to list returns" must {
 
       val cgtReference       = sample[CgtReference]
+      val today: LocalDate   = LocalDate.now()
       val (fromDate, toDate) =
-        LocalDate.now().minusYears(config.numberOfTaxYearsForReturns).plusDays(1L) -> LocalDate.now()
+        today.minusYears(config.numberOfTaxYearsForReturns).plusDays(1L) -> TimeUtils.getTaxYearEndDateInclusive(today)
```
Comment (LAJW) [inline test/uk/gov/hmrc/cgtpropertydisposalsfrontend/services/returns/ReturnsServiceImplSpec.scala:1880]: This is getting silly. I know you didn't write this, but this tuple is unnecessary.

```suggestion
        val fromDate = today.minusYears(config.numberOfTaxYearsForReturns).plusDays(1L)
        val toDate = TimeUtils.getTaxYearEndDateInclusive(today)
```

**Review by LAJW — APPROVED** [review]
A couple nitpicks. It'd be nice if they were addressed, but we can merge if we're in a rush.

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/services/returns/ReturnsService.scala (hunk 1 above)
Comment (nubz) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/services/returns/ReturnsService.scala:617]: done

**Review by LAJW — APPROVED** [review]

**Pull request merged**